### PR TITLE
Compress assembly files

### PR DESCRIPTION
**Pull request by d4straub**
Large assembly files are compressed and exported to the reulst folder. This can reduce the result folder by several GB. This is the first bullet point in #34.

## PR checklist
 - [x] This comment contains a description of changes (with reason)
 - [ ] If you've fixed a bug or added code that should be tested, add tests!
 - [ ] If necessary, also make a PR on the [nf-core/mag branch on the nf-core/test-datasets repo]( https://github.com/nf-core/test-datasets/pull/new/nf-core/mag)
 - [x] Ensure the test suite passes (`nextflow run . -profile test,docker`).
 - [x] Make sure your code lints (`nf-core lint .`).
 - [x] Documentation in `docs` is updated
 - [x] `CHANGELOG.md` is updated
 - [x] `README.md` is updated

### CODE REVIEW

**Comment by d4straub** [comment]
Works now, but I realized there are far too many errors ignored for testing purposes, e.g. also for process spades and spadeshybrid.

File: main.nf [hunk 1]
```diff
@@ -859,14 +859,18 @@ process krona {
 process megahit {
     tag "$name"
     publishDir "${params.outdir}/", mode: 'copy',
-        saveAs: {filename -> filename.indexOf(".fastq.gz") == -1 ? "Assembly/$filename" : null}
+        saveAs: {filename -> 
+          if (filename.indexOf(".gz") == -1 && filename.indexOf(".contigs.fa") == -1 ) "Assembly/$filename"
+          else if (filename.indexOf(".contigs.fa.gz") >0) "Assembly/$filename"
+          else null}
```
Comment (skrakau) [inline main.nf:865]:
```suggestion
        saveAs: {filename -> 
          if (filename.indexOf(".log") > 0 || filename.indexOf(".contigs.fa.gz") > 0 ) "Assembly/$filename"
          else null}
```
would this be an alternative?

File: main.nf [hunk 2]
```diff
@@ -890,16 +895,20 @@ process megahit {
 process spadeshybrid {
     tag "$id"
     publishDir "${params.outdir}/", mode: 'copy', pattern: "${id}*",
-        saveAs: {filename -> filename.indexOf(".fastq.gz") == -1 ? "Assembly/SPAdesHybrid/$filename" : null}
+        saveAs: {filename -> 
+          if (filename.indexOf(".fastq.gz") == -1  && filename.indexOf("_scaffolds.fasta") == -1) "Assembly/SPAdesHybrid/$filename"
+          else if (filename.indexOf("_scaffolds.fasta.gz") > 0) "Assembly/SPAdesHybrid/$filename"
+          else null}
```
Comment (skrakau) [inline main.nf:901]: same as above

File: main.nf [hunk 3]
```diff
@@ -918,23 +927,29 @@ process spadeshybrid {
...
 
 process spades {
     tag "$id"
     publishDir "${params.outdir}/", mode: 'copy', pattern: "${id}*",
-        saveAs: {filename -> filename.indexOf(".fastq.gz") == -1 ? "Assembly/SPAdes/$filename" : null}
-
+        saveAs: {filename -> 
+          if (filename.indexOf(".fastq.gz") == -1  && filename.indexOf("_scaffolds.fasta") == -1) "Assembly/SPAdes/$filename"
+          else if (filename.indexOf("_scaffolds.fasta.gz") > 0) "Assembly/SPAdes/$filename"
+          else null}
```
Comment (skrakau) [inline main.nf:943]: same as above

File: CHANGELOG.md [hunk 4]
```diff
@@ -10,6 +10,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - Add host read removal with Bowtie 2 and according custom section to MultiQC
 - Add separate MultiQC section for FastQC after preprocessing
 - Add social preview image
+- Compress asseembly files
```
Comment (skrakau) [inline CHANGELOG.md:13]:
```suggestion
- Compress assembly files
```

File: main.nf (hunk 1 above)
Comment (d4straub) [reply main.nf:865]: Yes, this certainly looks cleaner.

File: main.nf (hunk 2 above)
Comment (d4straub) [reply main.nf:900]: Probably not the same, but
```
saveAs: {filename -> 
          if (filename.indexOf(".log") > 0 || filename.indexOf("_scaffolds.fasta.gz") > 0 || filename.indexOf("_graph.gfa.gz") > 0 || filename.indexOf("_contigs.fasta.gz") > 0 ) "Assembly/$filename"
          else null}
```
would be better as well.

**Comment by skrakau** [comment]
The errors you mentioned above are independent of this PR, right?

File: main.nf (hunk 2 above)
Comment (d4straub) [reply main.nf]: I changed that.

File: main.nf (hunk 3 above)
Comment (d4straub) [reply main.nf]: I changed that.

**Comment by d4straub** [comment]
Yes.

**Pull request merged**